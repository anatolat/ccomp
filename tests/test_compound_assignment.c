

int glob_int;
char glob_char;

void test_local_int() {
	printf("test_local_int\n");
	int a = 10;
	printf("%d ", a += 1);
	printf("%d ", a -= 1);
	printf("%d ", a /= 2);
	printf("%d ", a *= 2);
	printf("%d ", a %= 3);
	
	printf("\n");
}

void test_local_char() {
	printf("test_local_char\n");
	char x = 13; // unused
	char a = 10;
	char b = 11; // unused
	printf("%d ", a += 1);
	printf("%d ", a -= 1);
	printf("%d ", a /= 2);
	printf("%d ", a *= 2);
	printf("%d ", a %= 3);
	
	printf("\n");
}

void test_glob_int() {
	printf("test_glob_int\n");
	glob_int = 10;
	printf("%d ", glob_int += 1);
	printf("%d ", glob_int -= 1);
	printf("%d ", glob_int /= 2);
	printf("%d ", glob_int *= 2);
	printf("%d ", glob_int %= 3);
	
	printf("\n");
}


void test_glob_char() {
	printf("test_glob_char\n");
	glob_char = 10;
	printf("%d ", glob_char += 1);
	printf("%d ", glob_char -= 1);
	printf("%d ", glob_char /= 2);
	printf("%d ", glob_char *= 2);
	printf("%d ", glob_char %= 3);
	
	printf("\n"); 
}

int main() {
	test_local_int();
	test_local_char();
	test_glob_int();
	test_glob_char();
}