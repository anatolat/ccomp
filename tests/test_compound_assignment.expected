test_local_int
11 10 5 10 1 
test_local_char
11 10 5 10 1 
test_glob_int
11 10 5 10 1 
test_glob_char
11 10 5 10 1 
